src/ProxyClient/HttpProxyClient.php

```diff
@@ -114,4 +114,47 @@ protected function escapeTags(array $tags)
 
         return $tags;
     }
+
+    /**
+     * Splits a header value into an array of values. E.g. useful for tag
+     * invalidation requests where you might need multiple requests if you
+     * want to invalidate too many cache tags (so the header would get too long).
+     *
+     * @param string $value
+     * @param int    $length
+     * @param string $delimiter
+     *
+     * @return array
+     */
+    protected function splitLongHeaderValue($value, $length = 7500, $delimiter = ',')
```

Comment:
finally looking over this again and i notice that this method is quite different from what we do to solve the same problem in Varnish::invalidateTags. can we instead put the logic from Varnish::invalidateTags into a HttpProxyClient::generateTagInvalidationChunks and use that for both?

this method is inefficient because the caller first implodes all tags, then splits them again and builds chunks from it